.github/workflows/aiohttp.yml

```
@@ -0,0 +1,63 @@
+name: Aiohttp
+
+on:
+  push:
+    branches:
+      - 'main'
+  pull_request:
+    branches:
+      - 'main'
+
+jobs:
+  test:
+    permissions:
+      contents: read # to fetch code (actions/checkout)
+
+    name: Aiohttp regression tests
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout aiohttp
+      uses: actions/checkout@v4
+      with:
+        repository: aio-libs/aiohttp
+    - name: Checkout llhttp
+      uses: actions/checkout@v4
+      with:
+        path: vendor/llhttp
+    - name: Restore node_modules cache
+      uses: actions/cache@v3
+      with:
+        path: vendor/llhttp/.npm
+        key: ubuntu-latest-node-${{ hashFiles('vendor/llhttp/**/package-lock.json') }}
+        restore-keys: ubuntu-latest-node-
+    - name: Install llhttp dependencies
+      run: npm install --ignore-scripts
+      working-directory: vendor/llhttp
+    - name: Build llhttp
+      run: make generate-llhttp
+    - name: Setup Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: 3.x
+        cache: 'pip'
+        cache-dependency-path: 'requirements/*.txt'
+    - name: Install aiohttp dependencies
+      uses: py-actions/py-dependency-install@v4
+      with:
+        path: requirements/test.txt
+    - name: Install aiohttp build dependencies
+      uses: py-actions/py-dependency-install@v4
+      with:
+        path: requirements/cython.txt
+    - name: Build aiohttp parser
+      run: make cythonize
+    - name: Install aiohttp
+      run: pip install .
+    - name: Run tests
+      env:
+        COLOR: yes
+      run: pytest tests/test_http_parser.py tests/test_web_functional.py
```

This https://github.com/aio-libs/aiohttp/blob/b51d5d7/.github/workflows/ci-cd.yml#L228C20-L231C18 modifies `$PATH` before invoking `make test` and here https://github.com/aio-libs/aiohttp/blob/b51d5d7/Makefile#L92C23-L92C61, there's another `pip install` with editable and constraints. So maybe try reproducing this via
```suggestion
    - name: Provision the dev env
      env:
        COLOR: yes
      run: >-
        PATH="${HOME}/.local/bin:${PATH}"
        make .develop
    - name: Run tests
      env:
        COLOR: yes
      run: >-
        PATH="${HOME}/.local/bin:${PATH}"
        pytest tests/test_http_parser.py tests/test_web_functional.py
```